mitosheet/mitosheet/api/graph/plotly_express_graphs.py

```diff
@@ -0,0 +1,213 @@
+    """
+    # Check if we filter the graph
+    _, filtered = filter_df_to_safe_size(graph_type, df, column_headers)
+    
+    # If we don't filter the graph, then return an empty string
+    if not filtered: 
+        return ''
+    
+    # If we do filter the graph, then return the code needed to filter the graph
+    filtered_code = """from mitosheet import filter_df_to_safe_size
+
+# Filter the dataframe so that it does not crash the browser
+{df_name}_filtered, _ = filter_df_to_safe_size('{graph_type}', {df_name}, {column_headers})
+""".format(
+    graph_type=graph_type,
+    df_name=df_name, 
+    column_headers=column_headers
+)
+    return filtered_code
+
+
+def graph_creation(
+    graph_type: str, 
+    df: pd.DataFrame, 
+    x_axis_column_headers: List[ColumnHeader],
+    y_axis_column_headers: List[ColumnHeader]
+) -> go.Figure: 
+    """
+    Creates and returns the Plotly express graph figure
+    """
+    num_x_axis_column_headers = len(x_axis_column_headers)
+    num_y_axis_column_headers = len(y_axis_column_headers)
+    
+    # Create the parameters that we use to construct the graph
+    x_arg = None
+    if num_x_axis_column_headers == 1:
+        # Note: In the new interface, x will always have a length of 0 or 1
+        x_arg = x_axis_column_headers[0]
+    if num_x_axis_column_headers > 1:
+        x_arg = x_axis_column_headers
+
+    y_arg = None
+    if num_y_axis_column_headers == 1:
+        y_arg = y_axis_column_headers[0]
+    if num_y_axis_column_headers > 1:
+        y_arg = y_axis_column_headers
+    
+    if graph_type == BOX:
+        return px.box(df, x=x_arg, y=y_arg)
+    if graph_type == HISTOGRAM:
+        return px.histogram(df, x=x_arg, y=y_arg)
+    if graph_type == BAR:
+        return px.bar(df, x=x_arg, y=y_arg)
+    if graph_type == SCATTER:
+        return px.scatter(df, x=x_arg, y=y_arg)
+
+    
+def graph_creation_code(
+    graph_type: str, 
+    df_name: str, 
+    x_axis_column_headers: List[ColumnHeader],
+    y_axis_column_headers: List[ColumnHeader]
+) -> str:
+    """
+    Returns the code for creating the Plotly express graph
+    """
+    num_x_axis_column_headers = len(x_axis_column_headers)
+    num_y_axis_column_headers = len(y_axis_column_headers)
+
+    # Create the chords that we use to construct the graph
+    x_chord = ''
+    if num_x_axis_column_headers == 1:
+        # Note: In the new interface, x will always have a length of 0 or 1
+        x_chord = f'x="{x_axis_column_headers[0]}"'
+    if num_x_axis_column_headers > 1:
+        x_chord = f'x={x_axis_column_headers}'
+
+    y_chord = ''
+    if num_y_axis_column_headers == 1:
+        y_chord = f'y="{y_axis_column_headers[0]}"'
+    if num_y_axis_column_headers > 1:
+        y_chord = f'y={y_axis_column_headers}'
+
+    # Create the string of all the parameters used to create the graph
+    all_chords = [df_name, x_chord, y_chord]
+    params = (', ').join(list(filter(lambda chord: chord != '', all_chords)))
```

Comment:
Another way of doing this that leads to more readable code (specifically that avoids default strings that are empty, which are usually a gross sign is something like the following:


```
all_params = dict()

if len(x_axis_column_headers) == 1:
     all_params['x'] = transpiled_column_header(x_axis_column_headers[0])
elif len(x_axis_column_headers) >= 1:
     all_params['x'] = transpiled_column_header_list_idk_func_name(x_axis_column_headers)

... same for x

params = ', '.join(f'{key}={value}' for key, value in all_params.items())
```

There are a couple of benefits, but to highlight a few:
1. we avoid creating temporary variables entirely (e.g. y_chord doesn't exist), which means less code. 
2. when we add _new_ params, it's easier. You just have to write the code that creates it (and you don't have to remember it to add to some `all_chords` list.
3. no nasty empty string variables! Empty string variables almost always demonstrate there's a better way to express things (b/c you can think about empty strings as encoding an "invalid" state without being in the type system.